### PR TITLE
Provide minimal padding for inside tick labels case of bar, heatmap, line plots

### LINKED ISSUE #5321: strange ticklabelposition behaviour
In this pen: https://codepen.io/nicolaskruchten/pen/PoGzzbK

1. the x range doesn't leave room for the y labels so they overlap the data
2. the y range expands seemingly more than necessary for the x labels

### LINKED ISSUE #5322: ticklabelposition inside with bars
Seems like we don't alter x range for y labels when the trace is a bar? https://codepen.io/nicolaskruchten/pen/bGweeoM

### PR DESCRIPTION
**Pull request by archmoj**
Fixes #5322 and fixes #5321 (padding for line plots & unnecessary padding).

Demo1: `bar`
[Before](https://codepen.io/MojtabaSamimi/pen/oNzLYLJ) vs [After](https://codepen.io/MojtabaSamimi/pen/QWKEGKG)

Demo2: `line plot`
[Before](https://codepen.io/MojtabaSamimi/pen/vYXKyXq) vs [After](https://codepen.io/MojtabaSamimi/pen/WNGxooo)

Demo3: `heatmap`
[Before](https://codepen.io/MojtabaSamimi/pen/poEbNwr) vs [After](https://codepen.io/MojtabaSamimi/pen/poEbNrr)

**TODO:**
- [x] add test

@plotly/plotly_js

### CODE REVIEW

File: src/plots/cartesian/autorange.js [hunk 1]
```diff
@@ -221,14 +221,18 @@ function makePadFn(ax, max) {
     extrappad = adjustPadForInsideLabelsOnAnchorAxis(extrappad, ax, max);
     extrappad = adjustPadForInsideLabelsOnThisAxis(extrappad, ax, max);
 
+    var pad0 = 0;
+    pad0 = adjustPadForInsideLabelsOnAnchorAxis(pad0, ax, max);
+    pad0 = adjustPadForInsideLabelsOnThisAxis(pad0, ax, max);
```
Comment (alexcjohnson) [inline src/plots/cartesian/autorange.js:226]: That'll work, but it seems like now the code is doing the same thing multiple times. It looks like both of these functions only use the first arg right at the end "return the greater of the newly-calculated value and this original padding." Can we keep the new lines and replace the two `extrappad = adjustPad...` lines above with just:
```js
extrappad = Math.max(extrappad, pad0);
```
after we finish calculating `pad0`?

**Comment by alexcjohnson** [comment]
@archmoj this looks great for including padding in the cases it was previously ignored. Do you want to sort out why the line case is getting such huge padding in the y direction in this PR, or do that in a separate PR?

File: src/plots/cartesian/autorange.js (hunk 1 above)
Comment (archmoj) [reply src/plots/cartesian/autorange.js]: Good call. Addressed in https://github.com/plotly/plotly.js/pull/5325/commits/f0709c319b84d813d82acd3d94afeb3d1709f670.

**Comment by archmoj** [comment]
> @archmoj this looks great for including padding in the cases it was previously ignored. Do you want to sort out why the line case is getting such huge padding in the y direction in this PR, or do that in a separate PR?

I'll try to address it in this PR.

**Comment by nicolaskruchten** [comment]
the heatmap case is a bit of an edge-case but this seems correct to me. I'm not sure what the point would be of putting the labels on the inside for a heatmap but if that's what they're asking for that's what they should get, I think. 

An eventual `<trace>.padding` prop could be used to override this, if they wanted inside ticks on top of the heatmap, presumably, correct?

**Review by alexcjohnson — APPROVED** [review]
Looks great! 💃

**Pull request merged**